### PR TITLE
feat: return number of remaining documents to process

### PR DESCRIPTION
**Pull request by y-lakhdar**
When user specifies the expected number of documents to process, success and error callbacks will be called with the remaining un-processed documents count.

### CODE REVIEW

File: src/interfaces.ts [hunk 1]
```diff
@@ -11,15 +11,26 @@ export interface BatchUpdateDocuments {
...
  *
  * @param {string[]} files Files from which the documentBuilders were generated
  * @param {DocumentBuilder[]} batch List of the uploaded DocumentBuilders
+ * @param {UploadProgress} progress Progress of the upload process
  * @param {AxiosResponse} res Axios response
  */
 export interface UploadBatchCallbackData {
   files: string[];
   batch: DocumentBuilder[];
+  progress?: UploadProgress;
```
Comment (y-lakhdar) [inline src/interfaces.ts:33]: Ideally, I would like the progress value to always be returned, but the ways the FileConsumer was designed (with the internal promise), it's hard to do so without a refactor.

Comment (louis-bompart) [reply src/interfaces.ts:33]: 👶 🐾  : that could be done in another PR

**Review by louis-bompart — CHANGES_REQUESTED** [review]
Try mocking the readFileSync method and assert on it instead of writing temporary files.

**Review by louis-bompart — APPROVED** [review]
Clean!

**Pull request merged**